### ISSUE TITLE
Fastlane 2.144.0 breaks compatibility

### ISSUE THREAD

**Issue by florianbuerger** [fastlane-community/fastlane-plugin-s3#84]
The [upgrade to the v3 AWS SDK](https://github.com/fastlane/fastlane/pull/16143) in [2.144.0](https://github.com/fastlane/fastlane/releases/tag/2.144.0) isn't compatible with this plugin. 

```
Bundler could not find compatible versions for gem "aws-sdk-core":
  In snapshot (Gemfile.lock):
    aws-sdk-core (= 3.92.0)

  In Gemfile:
    fastlane-plugin-aws_s3 was resolved to 1.6.1, which depends on
      aws-sdk (~> 2.3) was resolved to 2.11.451, which depends on
        aws-sdk-resources (= 2.11.451) was resolved to 2.11.451, which depends on
          aws-sdk-core (= 2.11.451)

    fastlane (~> 2.0) was resolved to 2.144.0, which depends on
      aws-sdk-s3 (~> 1.0) was resolved to 1.61.1, which depends on
        aws-sdk-core (~> 3, >= 3.83.0)
```

Seems this might be related →  [#16143](https://github.com/fastlane/fastlane/pull/16143#issuecomment-596610599)?

---

I can try to look into this later this week but I have no clue about ruby and the AWS SDK seems to be quite complicated on first look.

**Comment by cndouglas** [comment]
Might be fixed by #81.

**Comment by GauravP123** [comment]
getting following error 
`Bundler could not find compatible versions for gem "aws-sdk-core":
  In snapshot (Gemfile.lock):
    aws-sdk-core (= 3.92.0)

  In Gemfile:
    aws-sdk-elasticsearchservice was resolved to 1.32.0, which depends on
      aws-sdk-core (~> 3, >= 3.71.0)

    fastlane-plugin-aws_s3 was resolved to 1.6.1, which depends on
      aws-sdk (~> 2.3) was resolved to 2.3.0, which depends on
        aws-sdk-resources (= 2.3.0) was resolved to 2.3.0, which depends on
          aws-sdk-core (= 2.3.0)
`

**Comment by levibostian** [comment]
@florianbuerger, I encountered the same issue a while back. I made a PR to fix the problem. 

Try installing my forked repo as your fastlane plugin to use. In your `fastlane/Pluginfile`, add this line:

```
gem 'fastlane-plugin-aws_s3', git: 'https://github.com/levibostian/fastlane-plugin-s3/', ref: '8187839be3'
```

...and comment out your previous `gem 'fastlane-plugin-aws_s3'` line in the file which points to the original outdated plugin. 

---

Because you should never install someone's code just because they tell you to, [here](https://github.com/fastlane-community/fastlane-plugin-s3/pull/85/files) is the diff for my fork so you can read over the code I changed if you need to verify it.

**Comment by jeroendelau** [comment]
Unfortunately, this failed for me.
Fastlane version with 2.144

```
[16:06:22]: fastlane finished with errors
bundler: failed to load command: fastlane (/Users/jeroen/Development/clients/node/pocketchange/mobile/ios/aws/aws-sdk-core/ruby/2.6.0/bin/fastlane)
LoadError: cannot load such file -- aws-sdk
  /Users/jeroen/Development/clients/node/pocketchange/forkedlibs/fastlane-plugin-s3/lib/fastlane/plugin/aws_s3/actions/aws_s3_action.rb:85:in `require'
```

**Comment by levibostian** [comment]
Ah, thanks @jeroendelau for pointing that out. I realized that my iOS project I was testing with was using [the original](https://github.com/fastlane-community/fastlane-plugin-s3/pull/81) PR and not the modular one. I fixed the require crashed and pushed up an update to that branch.

**Comment by joshdholtz** [comment]
Hey all! I will get the new version out that fixes this this afternoon 😊 Sorry about the issue 😔

**Comment by GauravP123** [comment]
its fixed ? have to use `gem 'fastlane-plugin-aws_s3', git: 'https://github.com/levibostian/fastlane-plugin-s3/', ref: '3b517f92de' ?
`

**Comment by joshdholtz** [comment]
Fixed in version `1.170.0` 🚀
